hazelcast/src/hazelcast/client/compact.cpp

```diff
@@ -1489,6 +1489,8 @@ field_descriptor::field_descriptor(enum field_kind f,
 std::array<uint64_t, 256>
 init_fp_table()
 {
+    std::mutex tmp_mutex;
+    std::lock_guard<std::mutex> lg(tmp_mutex);
```

Comment:
Lock is not required, static initialization is already thread safe,

```suggestion
```

Reply:
I put it because of the static FP_TABLE initialization in init_fp_table. But maybe it will be better to remove both mutex and static keyword in this function.